zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -950,6 +916,70 @@ angular.module('zeppelinWebApp')
     }
   };
 
+  var getEditorSetting = function(interpreterName) {
+    var deferred = $q.defer();
+    websocketMsgSrv.getEditorSetting(interpreterName);
+    $timeout(
+      $scope.$on('editorSetting', function(event, data) {
+        deferred.resolve(data);
+      }
+    ), 1000);
+    return deferred.promise;
+  };
+
+  var setParagraphMode = function(session, paragraphText, pos) {
+    // Evaluate the mode only if the the position is undefined
+    // or the first 30 characters of the paragraph have been modified
+    // or cursor position is at beginning of second line.(in case user hit enter after typing %magic)
+    if ((typeof pos === 'undefined') || (pos.row === 0 && pos.column < 30) || (pos.row === 1 && pos.column === 0)) {
```

Comment:
similarly discussed here, should we just look for the first space after `%`?
https://github.com/apache/zeppelin/pull/1117

I think it will be good to get to be consistent in how we look for interpreter